spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/security/IapAuthenticationAutoConfiguration.java

```diff
@@ -59,7 +60,10 @@
 	@Bean
 	@ConditionalOnMissingBean
 	public JwtDecoder iapJwtDecoder(IapAuthenticationProperties properties) {
-		return new NimbusJwtDecoderJwkSupport(properties.getRegistry(), properties.getAlgorithm());
+		NimbusJwtDecoderJwkSupport jwkSupport
+				= new NimbusJwtDecoderJwkSupport(properties.getRegistry(), properties.getAlgorithm());
+		jwkSupport.setJwtValidator(JwtValidators.createDefaultWithIssuer(properties.getIssuer()));
```

Comment:
I think this `JwtValidator` has to be present as a bean, so you can provide something like `StubJwtValidator` for testing purpose to avoid such an issuer validation altogether as @meltsufin would like to do with an extra property.

Reply:
That's an interesting idea. So, I would write something like a NoopValidator, and optionally inject it from a sample app's configuration, to show how to do it? Or just provide NoopValidator without complicating the sample?
I am a bit concerned about allowing end users to make their web app insecure too easily. Perhaps in the scenario of testing, the end user could provide a stub/noop NimbusJwtDecoderJwkSupport instead of just the claims validator; then the end user app is completely responsible for all JWK validation/security themselves.

Reply:
That's good point too, having the fact that you already do `@ConditionalOnMissingBean`.
So, our default impl is based on the built-in `JwtValidators.createDefaultWithIssuer` based on the provided issuer.
And if you don't like it in your app, you are feel free to provide your own `JwtDecoder` bean.

That's all: nothing to discuss any more! 😄